v2/car.go

```diff
@@ -41,12 +46,90 @@ type (
 	}
 )
 
-// Size gets the size of Header in number of bytes.
-func (h *Header) Size() int {
-	return HeaderBytesSize
+// WriteTo writes this characteristics to the given writer.
+func (c Characteristics) WriteTo(w io.Writer) (n int64, err error) {
+	wn, err := writeUint64To(w, c.Hi)
+	if err != nil {
+		return
+	}
+	n += wn
+	wn, err = writeUint64To(w, c.Lo)
+	if err != nil {
+		return
+	}
+	n += wn
+	return
 }
 
 // Size gets the size of Characteristics in number of bytes.
-func (c *Characteristics) Size() int {
+func (c Characteristics) Size() uint64 {
 	return CharacteristicsBytesSize
 }
+
+// NewHeader instantiates a new CAR v2 header, given the byte length of a CAR v1.
+func NewHeader(carV1Size uint64) *Header {
+	header := &Header{
+		Characteristics: EmptyCharacteristics,
+		CarV1Size:       carV1Size,
+	}
+	header.CarV1Offset = PrefixBytesSize + HeaderBytesSize
+	header.IndexOffset = header.CarV1Offset + carV1Size
+	return header
+}
+
+// Size gets the size of Header in number of bytes.
+func (h *Header) Size() uint64 {
+	return HeaderBytesSize
```

Comment:
I think you should either expose the global constant, or the method, but not both - that seems redundant. Given that the size is entirely static, I'd just remove the method. Otherwise it seems like the Header's size is dependent on some of its fields, given that it's a method.

Reply:
I was hoping to introduce a `Sizer` interface or something to that effect for all segments that make up a car. This is to make calculation of a CAR's size to be a series of `Size()` calls on segments. Happy to remove if you think this is redundant.

Reply:
If you prefer an interface-like system, that's also an option, but then I'd unexport the constants. You really don't want to expose multiple ways to do the same thing in the documentation unless strictly necessary, because that can confuse users and make it harder to start using the library.

I think I'd still lean on constants, because obtaining the header size via `Header{}.Size()` feels slightly roundabout. I also can't think of any advantage to the interface other than "it looks nice" :)